numba/_dispatcher.cpp

```diff
@@ -97,11 +98,12 @@ else                                                            \
     }                                                           \
 }
 
+typedef std::vector<Type> TypeTable;
+typedef std::vector<PyObject*> Functions;
 
-typedef struct DispatcherObject{
+class Dispatcher {
+public:
     PyObject_HEAD
-    /* Holds borrowed references to PyCFunction objects */
-    dispatcher_t *dispatcher;
     char can_compile;        /* Can auto compile */
     char can_fallback;       /* Can fallback */
     char exact_match_required;
```

Comment:
```suggestion
    char exact_match_required; /* Whether an exact match is required */
```

Reply:
I've added a slightly different description.

Reply:
Yours is better :) Thanks!